src/cacheUntilType.js

```diff
@@ -0,0 +1,51 @@
+import { OuterSubscriber } from 'rxjs/OuterSubscriber';
+import { subscribeToResult } from 'rxjs/util/subscribeToResult';
+
+
+export function cacheUntilType(type) {
+  return this.lift(new CacheUntilTypeOperator(type));
+}
+
+class CacheUntilTypeOperator {
```

Comment:
I _think_ this is an equivalent closure version, probably even smaller if you compared the compiled output of the class version. Don't need to use my logic, just providing for future reference in case it's not clear what I meant. (this isn't a pattern really documented anywhere)

```js
export function queueUntilType(expectedType) {
  return this.lift(function (source) {
    let queue = [];
    let hasSeenType = false;

    return source.subscribe({
      next: action => {
        // It's important to check this first because of possible re-entry
        // before we've finished flushing the queue.
        if (hasSeenType) {
          this.next(action);
          return;
        }

        if (action && action.type === expectedType) {
          hasSeenType = true;
          this.add(subscribeToResult(this, queue));
          // no longer needed, free memory
          queue = null;
        } else {
          queue.push(action);
        }
      },
      error: e => this.error(e),
      complete: () => this.complete()
    });
  });
}
```

***

I did the conditional differently but I believe it has the same effect. 😝  Your way is totally fine.